php/class-customize-snapshot-manager.php

```diff
@@ -0,0 +1,559 @@
+	 */
+	public function store_post_data() {
+		if ( current_user_can( 'customize' ) && isset( $_POST['snapshot_customized'] ) ) {
+			$this->post_data = json_decode( wp_unslash( $_POST['snapshot_customized'] ), true );
+
+			// Contextual settings.
+			foreach ( $this->post_data as $setting_id => $value ) {
+				$matches = self::parse_contextual_setting_id( $setting_id );
+				if ( isset( $matches['context'] ) ) {
+					// Remove contextuals from the post_data array.
+					unset( $this->post_data[ $setting_id ] );
+
+					// Global contextuals write over settings to fix scope creep caused by the plugin.
+					if ( 'global' === $matches['context'] ) {
+						$_setting_id = $matches['base'];
+						if ( isset( $matches['key'] ) ) {
+							$_setting_id .= $matches['key'];
+						}
+						$this->post_data[ $_setting_id ] = $value;
+					} else if ( ! empty( $matches['context'] ) ) {
+						$this->contextual_post_data[ $setting_id ] = $value;
+					}
+				}
+			}
+		}
+	}
+
+	/**
+	 * Create the custom post type.
+	 *
+	 * @access public
+	 */
+	public function create_post_type() {
+		$args = array(
+			'labels' => array(
+				'name' => __( 'Customize Snapshots', 'customize-widgets-plus' ),
+				'singular_name' => __( 'Customize Snapshot', 'customize-widgets-plus' ),
+			),
+			'public' => false,
+			'capability_type' => 'post',
+			'map_meta_cap' => true,
+			'hierarchical' => false,
+			'rewrite' => false,
+			'delete_with_user' => false,
+			'supports' => array( 'title', 'author', 'revisions' ),
+		);
+
+		register_post_type( self::POST_TYPE, $args );
+	}
+
+	/**
+	 * Enqueue styles & scripts for the Customizer.
+	 *
+	 * @action customize_controls_enqueue_scripts
+	 */
+	public function enqueue_scripts() {
+		$handle = 'customize-snapshot';
+
+		// Enqueue styles.
+		wp_enqueue_style( $this->plugin->style_handles[ $handle ] );
+
+		// Enqueue scripts.
+		wp_enqueue_script( $this->plugin->script_handles[ $handle ] );
+
+		// Set the sanpshot theme.
+		$snapshot_theme = null;
+		if ( isset( $this->snapshot->post()->ID ) ) {
+			$snapshot_theme = get_post_meta( $this->snapshot->post()->ID, '_snapshot_theme', true );
+		}
+
+		// Script data array.
+		$exports = array(
+			'nonce' => wp_create_nonce( self::AJAX_ACTION ),
+			'action' => self::AJAX_ACTION,
+			'uuid' => $this->snapshot->uuid(),
+			'is_preview' => $this->snapshot->is_preview(),
+			'snapshot_theme' => $snapshot_theme,
+			'scope' => ( isset( $_GET['scope'] ) ? $_GET['scope'] : 'dirty' ),
+			'i18n' => array(
+				'saveButton' => __( 'Save', 'customize-widgets-plus' ),
+				'cancelButton' => __( 'Cancel', 'customize-widgets-plus' ),
+				'shareButton' => __( 'Share URL to preview', 'customize-widgets-plus' ),
+				'updateMsg' => __( 'Clicking "Save" will update the current snapshot.', 'customize-widgets-plus' ),
+				'errorMsg' => __( 'The snapshot could not be saved.', 'customize-widgets-plus' ),
+				'previewTitle' => __( 'Preview URL', 'customize-widgets-plus' ),
```

Comment:
As noted in Slack, this isn't getting displayed after submitting the desired scope.